src/tools/tidy/src/ui_tests.rs

```diff
@@ -128,4 +132,44 @@ pub fn check(path: &Path, bad: &mut bool) {
             }
         }
     });
+
+    // Check that any test files modified don't have a useless name
+    for changed_file in git::get_git_modified_files(None, &vec!["rs"]).unwrap().unwrap() {
+        // `get_git_modified_files` reports all files in the repo, only check ui tests
+        if !paths.iter().any(|path| changed_file.starts_with(&path.to_string_lossy().into_owned()))
+        {
+            continue;
+        }
+
+        let changed_file = changed_file.as_str();
+
+        // use git rev-list to make sure that the file still exists, since `get_git_modified_files`
+        // names both the source and destination files in a rename.
+        let mut cmd = Command::new("git");
+        cmd.args(["rev-list", "-1", "--all"]);
+        cmd.arg(changed_file);
+
+        // If a non-zero status was returned, it's from a deleted file, otherwise it exists
+        let is_modified = cmd.output().expect("unable to execute `git rev-list`").status.success();
+
+        if is_modified {
+            lazy_static! {
+                static ref ISSUE_NAME_REGEX: Regex = Regex::new(r"issues?[-_]?\d+").unwrap();
```

Comment:
Sometimes people add a sequence of files with a deliberate numbering scheme, which I am slightly worried about catching simple false positives on of up to maybe 2 digits, whereas our oldest open issue has at least 4 digits. Also, we're testing this regex only on the file stem, so we can first make sure we're matching on the root:
```suggestion
                static ref ISSUE_NAME_REGEX: Regex = Regex::new(r"^issues?[-_]?\d{4,}").unwrap();
```

Reply:
We chatted a bit and this is further motivated by the fact that if real information is going to be in the test name, it's best that it not be sorted together with a pile of other meaningless `issue-\d+-actual-test-names` but rather that it go next to `actual-test-name-about-similar-things`.

Reply:
Some files use three digits:
- `tests/ui/generics/issue-333.rs`
- `tests/ui/closures/issue-868.rs`
- `tests/ui/reachable/issue-948.rs`
- `tests/ui/drop/issue-979.rs`
So I had to error on three or more digits.

Also when adding the `^` anchor, I found `tests/ui/feature-gates/unstable-attribute-allow-issue-0.rs`, which definitely shouldn't be forbidden at all, so I'm glad we talked about this!